packages/neo-one-node-bin/src/utils/getConfiguration.ts

```diff
@@ -0,0 +1,112 @@
+import { FullNodeEnvironment, FullNodeOptions as ConfigOptions } from '@neo-one/node';
+import { createMain, createTest } from '@neo-one/node-neo-settings';
+import chokidar from 'chokidar';
+import envPaths from 'env-paths';
+import fs from 'fs-extra';
+import rc from 'rc';
+import { BehaviorSubject, concat, Observable, of } from 'rxjs';
+import { scan } from 'rxjs/operators';
+import SeamlessImmutable from 'seamless-immutable';
+import { createMonitor, MonitorEnvironment } from './createMonitor';
+
+export interface NodeBinEnvironment {
+  readonly fullNodeEnvironment: FullNodeEnvironment;
+  readonly settingsEnvironment: SettingsEnvironment;
+  readonly monitorEnvironment: MonitorEnvironment;
+  readonly options: ConfigOptions;
+  // tslint:disable-next-line:readonly-array
+  readonly configs: string[] | undefined;
+}
+
+interface SettingsEnvironment {
+  readonly type: 'main' | 'test';
+  readonly address?: string;
+  readonly secondsPerBlock?: number;
+  readonly standbyValidators?: ReadonlyArray<string>;
+}
+
+const DEFAULT_CONFIG = {
+  environment: {
+    dataPath: envPaths('neo-one-node', { suffix: false }).data,
+    rpc: {
+      http: {
+        port: 8080,
+        host: 'localhost',
+      },
+    },
+  },
+  settings: {
+    type: 'test',
+  },
+  monitor: {
+    level: 'verbose',
+  },
+};
+
+const getSettings = (settings: SettingsEnvironment) => {
+  const { type, ...rest } = settings;
+  switch (type) {
+    case 'main': {
+      return createMain(rest);
+    }
+    case 'test': {
+      return createTest(rest);
+    }
+    default: {
+      throw new Error("`type` must be 'main' or 'test'.");
+    }
+  }
+};
+
+const getFreshConfig = (): NodeBinEnvironment => {
+  const config = rc('neo-one-node', DEFAULT_CONFIG);
```

Comment:
Probably better if it's `neo_one_node` since that's what the environment variables will use as a prefix.